### PR TITLE
Fixes #564 now susper answers questions

### LINKED ISSUE #564: Bring Intelligence to Susper
right now google answers to questions people ask. 

image

we could implement the same feature in susper using SUSI AI. this even helps to improve the use cases of SUSI.
image

### PR DESCRIPTION
**Pull request by nikhilrayaprolu**
Fixes issue #564 
This is basic implementation of Intelligence which answers only for questions. this could be incrementally developed with new features like displaying charts, maps etc. and UI changes could be done.
Demo Link: https://susper-pr-569.herokuapp.com/

Screenshots for the change: 
screenshot from 2017-06-28 23-45-49

### CODE REVIEW

File: src/app/intelligence.service.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export class IntelligenceService {
+  server = 'http://api.susi.ai';
+  searchURL = 'http://' + this.server + '/susi/chat.json';
+  constructor(private http: Http, private jsonp: Jsonp, private store: Store<fromRoot.State>) {
+  }
+  getintelligentresponse(searchquery) {
+    let params = new URLSearchParams();
+    params.set('q', searchquery);
+    // params.set('QueryClass', 'MaxHits=5');
+
```
Comment (harshit98) [inline src/app/intelligence.service.ts]: Remove the comment also. There is no need to add `QueryClass` and `MaxHits` in future. We won't be using them.

File: src/app/intelligence.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return this.jsonp
+      .get('http://api.asksusi.com/susi/chat.json', {search: params}).map(res =>
+        res.json()
+
+      );
+  }
+
+
+
```
Comment (harshit98) [inline src/app/intelligence.service.ts]: Single space only. You're giving too much space here.

File: src/app/intelligence/intelligence.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<div class="card" *ngIf="answer">
+{{this.answer}}
```
Comment (harshit98) [inline src/app/intelligence/intelligence.component.html]: indent it.

File: src/app/intelligence/intelligence.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  constructor(private store: Store<fromRoot.State>, private intelligence: IntelligenceService) {
+    this.wholequery$ = store.select(fromRoot.getwholequery);
+    this.wholequery$.subscribe(data => {
+      this.intelligence.getintelligentresponse(data.query).subscribe(res => {
+        console.log(res, res.answers , res.answers[0]);
+        if (res && res.answers && res.answers[0].actions) {
+          this.actions = res.answers[0].actions;
+          for (let action of this.actions){
+            if (action.type === 'answer' && action.mood !== 'sabta') {
+               this.answer = action.expression;
```
Comment (harshit98) [inline src/app/intelligence/intelligence.component.ts]: what is 'sabta' ?

File: src/app/intelligence/intelligence.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+})
+export class IntelligenceComponent implements OnInit {
+  wholequery$: Observable<any>;
+  actions: Array<any> = [];
+  answer: any;
+  constructor(private store: Store<fromRoot.State>, private intelligence: IntelligenceService) {
+    this.wholequery$ = store.select(fromRoot.getwholequery);
+    this.wholequery$.subscribe(data => {
+      this.intelligence.getintelligentresponse(data.query).subscribe(res => {
+        console.log(res, res.answers , res.answers[0]);
```
Comment (harshit98) [inline src/app/intelligence/intelligence.component.ts]: remove console.log to avoid console errors.

**Comment by harshit98** [comment]
@nikhilrayaprolu Awesome! :100: 

Kindly go through my requested changes. :)

File: src/app/intelligence/intelligence.component.ts (hunk 4 above)
Comment (nikhilrayaprolu) [inline src/app/intelligence/intelligence.component.ts]: this is some random mood name given in susi when the results are not sure, hence whenever mood is sabta, didn't display the result.

**Comment by nikhilrayaprolu** [comment]
@harshit98 made the requested changes.

**Comment by mariobehling** [comment]
Great job. I see travis is failing also on live site. Please help to solve it in subsequent PRs.

**Pull request merged**